### PR TITLE
[STORM-440] completed exposing the drpcclient without need for configs

### PR DESCRIPTION
**Pull request by redsanket**
completed exposing the drpcclient and nimbusclient without the need for config initialization

### CODE REVIEW

File: storm-core/src/jvm/backtype/storm/utils/DRPCClient.java [hunk 1]
```diff
@@ -46,7 +46,22 @@ public DRPCClient(Map conf, String host, int port, Integer timeout) throws TTran
...
+    return getConfiguredDRPCClient(null, host, port);
+    }
+
+    public static DistributedRPC.Client getConfiguredDRPCClient(Map conf, String host, int port) {
+      try {
+        Map fullConf = Utils.readStormConfig();
+        if (conf != null) {
+          fullConf.putAll(conf);
+        }
+        return new DRPCClient(conf, host, port).getClient();
```
Comment (caofangkun) [inline storm-core/src/jvm/backtype/storm/utils/DRPCClient.java]: shoud be ```return new DRPCClient(fullConf, host, port).getClient();```

File: storm-core/src/jvm/backtype/storm/utils/NimbusClient.java [hunk 2]
```diff
@@ -30,11 +30,18 @@
...
-        try {
-            String nimbusHost = (String) conf.get(Config.NIMBUS_HOST);
-            return new NimbusClient(conf, nimbusHost);
+      try {
+        Map fullConf = Utils.readStormConfig();
+        if (conf != null) {
+         fullConf.putAll(conf);
+        }
+        String nimbusHost = (String) conf.get(Config.NIMBUS_HOST);
+              return new NimbusClient(conf, nimbusHost);
```
Comment (caofangkun) [inline storm-core/src/jvm/backtype/storm/utils/NimbusClient.java]: should be 
```
 String nimbusHost = (String) fullConf.get(Config.NIMBUS_HOST);
 return new NimbusClient(fullConf, nimbusHost);
```

Comment (redsanket) [inline storm-core/src/jvm/backtype/storm/utils/NimbusClient.java]: Sorry pushed the wrong commit, shall do it now, thanks for pointing it out

**Comment by ppoulosk** [comment]
+1 NB

File: storm-core/src/jvm/backtype/storm/utils/NimbusClient.java [hunk 3]
```diff
@@ -30,11 +30,18 @@
...
-        try {
-            String nimbusHost = (String) conf.get(Config.NIMBUS_HOST);
-            return new NimbusClient(conf, nimbusHost);
+      try {
+        Map fullConf = Utils.readStormConfig();
+        if (conf != null) {
+         fullConf.putAll(conf);
+        }
+        String nimbusHost = (String) fullConf.get(Config.NIMBUS_HOST);
+              return new NimbusClient(fullConf, nimbusHost);
```
Comment (revans2) [inline storm-core/src/jvm/backtype/storm/utils/NimbusClient.java]: very minor, spacing issue.

**Comment by revans2** [comment]
+1 needs to be upmerged though.

**Comment by harshach** [comment]
@redsanket can you upmerge and squash those commits.

**Comment by redsanket** [comment]
@harshach updated and squashed thank you

**Comment by harshach** [comment]
+1

**Pull request closed without merging**